host/crypto/openssl/cert.c

```diff
@@ -669,6 +669,8 @@ oe_result_t oe_cert_chain_read_pem(
     result = OE_OK;
 
 done:
+    if (result != OE_OK && sk)
+        sk_X509_pop_free(sk, X509_free);
```

Comment:
Is there a reason why this leak had gone undetected? I'd suggest adding a test that would lock down the scenario.

Reply:
We didn't enable debugmalloc or similar features on the host. The memory leaks in the PR are found because I was porting the openssl wrapper to the enclave where debugmalloc is enabled.

Reply:
Ah Ok. Thanks for the clarification. 
Do you know if it is feasible to enable valgrind or something else to detect host side memory issues?

Reply:
I think either valgrind or clang ([LeakSanitizer](https://clang.llvm.org/docs/LeakSanitizer.html)) should just work.

Reply:
> I think either valgrind or clang ([LeakSanitizer](https://clang.llvm.org/docs/LeakSanitizer.html)) should just work.

Created issue #3298